libvast/vast/format/json.hpp

```diff
@@ -147,6 +152,8 @@ class reader final : public multi_layout_reader {
   mutable size_t num_invalid_lines_ = 0;
   mutable size_t num_unknown_layouts_ = 0;
   mutable size_t num_lines_ = 0;
+  vast::duration read_timeout_
```

Comment:
To me this is attribute of every reader independent of the format, so we should add it there. If you do this, please leave the parsing from the options where it is and only pass the `duration` to the base class.